app/controllers/KairosDbProxyController.java

```diff
@@ -173,6 +184,35 @@ public KairosDbProxyController(
         }
     }
 
+    /* package private */ MetricsQuery checkAndAddMergeAggregator(final MetricsQuery metricsQuery) {
+        final List<Metric> newMetrics = new ArrayList<>();
+        for (final Metric metric : metricsQuery.getMetrics()) {
+            if (needMergeAggregator(metric.getAggregators())) {
+                final List<Aggregator> newAggregators = new ArrayList<>();
+                final Optional<Aggregator> aggregatorWithSampling = metric.getAggregators().stream().filter(
+                        aggregator -> aggregator.getSampling().isPresent()).findFirst();
+                newAggregators.add(aggregatorWithSampling.map(
```

Comment:
If the aggregator has parameters (e.g. percentile value) that don't apply to the merge aggregator, we should probably clear them. My point here, is that I suspect it is better to construct a new MergeAggregator definition than to clone from a random target one just to get its sampling.